### ISSUE TITLE
Debug 1.10 causes No child processes (Errno::ECHILD) at end of Rails test

### ISSUE THREAD

**Issue by ccasabona** [ruby/debug#290]
Running any of these commands:

```
Rails test
Rails test:models
```

causes Errno:ECHILD exception at the end of the testing.  

```
site [master-rails7●] % rails test:models
Running 826 tests in parallel using 4 processes
DEBUGGER: Detaching after fork from parent process 10222
DEBUGGER: Attaching after process 10222 fork to child process 10230
Run options: --seed 30315

# Running:

..........................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
/Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `waitpid': No child processes (Errno::ECHILD)
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `block (2 levels) in fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:26:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1626:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:26:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization/worker.rb:15:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:37:in `block in start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `times'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `each'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `map'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelize_executor.rb:18:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/minitest-5.14.4/lib/minitest.rb:138:in `run'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/minitest-5.14.4/lib/minitest.rb:68:in `block in autorun'
/Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `waitpid': No child processes (Errno::ECHILD)
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `block (2 levels) in fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:26:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1626:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:26:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization/worker.rb:15:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:37:in `block in start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `times'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `each'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `map'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelize_executor.rb:18:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/minitest-5.14.4/lib/minitest.rb:138:in `run'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/minitest-5.14.4/lib/minitest.rb:68:in `block in autorun'
/Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `waitpid': No child processes (Errno::ECHILD)
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `block (2 levels) in fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:26:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1626:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:8:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/fork_tracker.rb:26:in `fork'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization/worker.rb:15:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:37:in `block in start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `times'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `each'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `map'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelization.rb:36:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/activesupport-7.0.0.alpha2/lib/active_support/testing/parallelize_executor.rb:18:in `start'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/minitest-5.14.4/lib/minitest.rb:138:in `run'
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/minitest-5.14.4/lib/minitest.rb:68:in `block in autorun'


Finished in 8.734624s, 94.5662 runs/s, 773.3590 assertions/s.
826 runs, 6755 assertions, 0 failures, 0 errors, 0 skips
/Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `waitpid': No child processes (Errno::ECHILD)
	from /Users/charlie/.rvm/gems/ruby-3.0.2@ruby3rails7/gems/debug-1.1.0/lib/debug/session.rb:1646:in `block (2 levels) in fork'

```
This does not occur when running a single test, as in 

`rails test test/models/cart_test.rb`

This was not a problem with Debug 1.0.0.

Rails Version: 7.0.0.alpha and 7.0.0.alpha2

Ruby Version:  3.0.2

**Comment by st0012** [comment]
@ccasabona can you try the latest master? I think it's fixed by #280.

**Comment by ccasabona** [comment]
You are correct.  The latest master fixes this problem.
